### PR TITLE
refactor: add GitPluginClient

### PR DESCRIPTION
**Pull request by l-qing**
# Changes
* refactor: add GitPluginClient

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [ ] [`spec` PR link](https://github.com/katanomi/spec) included
- [x] Follows the [commit message standard](https://github.com/katanomi/spec/blob/main/CONTRIBUTING.md#commits)
- [x] Meets the [contributing guidelines](https://github.com/katanomi/spec/blob/main/CONTRIBUTING.md) (including
  functionality, content, code)
- [x] Test cases with documentation and functionality works as expected using current and related github repos (MUST deploy and check)
- [x] Release notes block below has been filled in or deleted (only if no user facing changes)

### CODE REVIEW

File: plugin/client/plugin_client.go [hunk 1]
```diff
@@ -127,21 +118,11 @@ func (p *PluginClient) WithClassAddress(classAddress *duckv1.Addressable) *Plugi
 	return p
 }
 
-func (p *PluginClient) WithClassObject(object ctrlclient.Object) *PluginClient {
-	p.ClassObject = object
```
Comment (l-qing) [inline plugin/client/plugin_client.go:131]: 这个从 PluginClient 移出去了。
是因为 赋值可能是在类似 GitPluginClient 中，如果要赋值，还需要再在 GitPluginClient 中写一次，没太大必要。

File: plugin/client/git_plugin_client.go [hunk 2]
```diff
@@ -25,19 +25,52 @@ import (
...
+	ClassObject ctrlclient.Object
+}
+
+// Clone shallow clone the plugin client
+// used to update some fields without changing the original
+func (p *GitPluginClient) Clone() *GitPluginClient {
+	if p == nil {
+		return nil
+	}
+	p = &(*p)
```
Comment (nanjingfm) [inline plugin/client/git_plugin_client.go:53]: 考虑下真实使用的场景，GitRepo和ClassObject需要copy么？

**Review by nanjingfm — DISMISSED** [review]
看下评论，先approve了

File: plugin/client/git_plugin_client.go (hunk 2 above)
Comment (l-qing) [reply plugin/client/git_plugin_client.go:53]: 也是

**Review by nanjingfm — APPROVED** [review]

**Pull request merged**